astroquery/solarsystem/pds/core.py

```diff
@@ -0,0 +1,447 @@
+# commonly required local imports shown below as example
+# all Query classes should inherit from BaseQuery.
+from ...query import BaseQuery
+from ...utils import async_to_sync
+
+# import configurable items declared in __init__.py, e.g. hardcoded dictionaries
+from . import conf
+
+
+@async_to_sync
+class RingNodeClass(BaseQuery):
+    """
+    for querying the Planetary Ring Node ephemeris tools
+    <https://pds-rings.seti.org/tools/>
+
+
+    # basic query for all six targets
+    for major_body in ['mars', 'jupiter', 'uranus', 'saturn', 'neptune', 'pluto']:
+        nodequery = RingNode(major_body, '2022-05-03 00:00')
+        systemtable, bodytable, ringtable = nodequery.ephemeris()
+
+        print(' ')
+        print(' ')
+        print('~'*40)
+        print(major_body)
+        print('~'*40)
+        print(systemtable)
+        print(bodytable)
+        print(ringtable)
+
+
+    """
+
+    TIMEOUT = conf.timeout
+
+    def __init__(self, planet=None, obs_time=None):
+        """Instantiate Planetary Ring Node query
+
+        Parameters
+        ----------
+        planet : str, required. one of Jupiter, Saturn, Uranus, or Neptune
+            Name, number, or designation of the object to be queried.
+        obs_time : str, in JD or MJD format. If no obs_time is provided, the current
+            time is used.
+        """
+
+        super().__init__()
+        self.planet = planet
+        self.obs_time = obs_time
+
+    def __str__(self):
+        """
+        String representation of RingNodeClass object instance'
+
+        Examples
+        --------
+        >>> from astroquery.solarsystem.pds import RingNode
+        >>> uranus = RingNode(planet='Uranus',
+        ...                 obs_time='2017-01-01 00:00')
+        >>> print(uranus)  # doctest: +SKIP
+        PDSRingNode instance "Uranus"; obs_time='2017-01-01 00:00'
+        """
+        return ('PDSRingNode instance "{:s}"; obs_time={:s}').format(
+            str(self.planet), str(self.obs_time)
+        )
+
+    # --- pretty stuff above this line, get it working below this line ---
+
+    def ephemeris_async(
+        self,
+        observer_coords=None,
+        neptune_arcmodel=3,
+        get_query_payload=False,
+        get_raw_response=False,
+        cache=True,
+    ):
+        """
+        send query to server
+
+        note this interacts with utils.async_to_sync to be called as ephemeris()
+
+        Parameters
+        ----------
+        self : RingNodeClass instance
+        observer_coords : three-element list/array/tuple of format (lat (deg), lon (deg east), altitude (m))
+
+        Returns
+        -------
+        response : `requests.Response`
+            The response of the HTTP request.
+
+        Examples
+        --------
+        >>> from astroquery.solarsystem.pds import RingNode
+        >>> uranus = RingNode(planet='Uranus',
+        ...                 obs_time='2017-01-01 00:00')
+        >>> eph = obj.ephemeris()  # doctest: +SKIP
```

Comment:
By assigning `RingNode(...)` to `uranus`, this example becomes a little bit muddied.  I would think that `obs_time` is more closely aligned with the ephemeris() parameters, rather than the planet itself.